Sources/ArcGISToolkit/Components/JobManager/JobManager.swift

```diff
@@ -0,0 +1,304 @@
+    
+    /// The background task identifier for status checks.
+    private let statusChecksTaskIdentifier = "com.esri.ArcGISToolkit.jobManager.statusCheck"
+    
+    // A Boolean value indicating whether a background status check is scheduled.
+    private var isBackgroundStatusChecksScheduled = false
+    
+    /// An initializer for the job manager.
+    private init() {
+        NotificationCenter.default.addObserver(self, selector: #selector(appWillEnterForeground), name: UIApplication.willEnterForegroundNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(appWillResignActive), name: UIApplication.willResignActiveNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(appWillTerminate), name: UIApplication.willTerminateNotification, object: nil)
+        
+        BGTaskScheduler.shared.register(forTaskWithIdentifier: statusChecksTaskIdentifier, using: nil) { task in
+            // Reset flag because once the task is launched, we need to reschedule if we want to do
+            // another background task.
+            self.isBackgroundStatusChecksScheduled = false
+            Task {
+                Logger.jobManager.debug("Performing status checks.")
+                await self.performStatusChecks()
+                Logger.jobManager.debug("Status checks completed.")
+                self.scheduleBackgroundStatusCheck()
+                task.setTaskCompleted(success: true)
+            }
+        }
+        
+        // Load jobs from the saved state.
+        loadState()
+    }
+    
+    /// Schedules a status check in the background if one is not already scheduled.
+    private func scheduleBackgroundStatusCheck() {
+        // Return if already scheduled.
+        guard !isBackgroundStatusChecksScheduled else {
+            return
+        }
+        
+        // Do not schedule if there are no running jobs.
+        guard hasRunningJobs else {
+            return
+        }
+        
+        // Make sure the preferred background status check schedule
+        guard case .regularInterval(let timeInterval) = preferredBackgroundStatusCheckSchedule else {
+            return
+        }
+        
+        isBackgroundStatusChecksScheduled = true
+        
+        let request = BGAppRefreshTaskRequest(identifier: statusChecksTaskIdentifier)
+        request.earliestBeginDate = Calendar.current.date(byAdding: .second, value: Int(timeInterval), to: .now)
+        do {
+            try BGTaskScheduler.shared.submit(request)
+            Logger.jobManager.debug("Background task scheduled.")
+        } catch(let error) {
+            Logger.jobManager.error("Background task scheduling error \(error.localizedDescription)")
+        }
+    }
+    
+    /// A Boolean value indicating if there are jobs running.
+    private var hasRunningJobs: Bool {
+        !jobs.filter({ $0.status == .started }).isEmpty
```

Comment:
Can this just check for any with `.started`?

```suggestion
        jobs.contains { $0.status == .started }
```